### PR TITLE
En-5761 : remove tx sign public and private keys

### PR DESCRIPTION
**Pull request by miiu96**

### CODE REVIEW

File: node/nodeTesting.go [hunk 1]
```diff
@@ -181,12 +184,13 @@ func (n *Node) generateAndSignSingleTx(
 	rcvAddrBytes []byte,
 	sndAddrBytes []byte,
 	data string,
+	sk crypto.PrivateKey,
 ) (*transaction.Transaction, []byte, error) {
 
 	if n.marshalizer == nil {
```
Comment (iulianpascalau) [inline node/nodeTesting.go]: check if nil here and the following lines

File: statusHandler/view/termuic/termuiRenders/widgetsRender.go [hunk 2]
```diff
@@ -143,16 +140,16 @@ func (wr *WidgetsRender) prepareInstanceInfo() {
 		consensusInfo = fmt.Sprintf("Consensus accepted / signed blocks: %d / %d", countConsensusAcceptedBlocks, countConsensus)
 	}
 
-	rows[4] = []string{consensusInfo}
+	rows[3] = []string{consensusInfo}
 
 	countLeader := wr.presenter.GetCountLeader()
 	countAcceptedBlocks := wr.presenter.GetCountAcceptedBlocks()
-	rows[5] = []string{fmt.Sprintf("Blocks accepted / blocks proposed : %d / %d", countAcceptedBlocks, countLeader)}
+	rows[4] = []string{fmt.Sprintf("Blocks accepted / blocks proposed : %d / %d", countAcceptedBlocks, countLeader)}
```
Comment (iulianpascalau) [inline statusHandler/view/termuic/termuiRenders/widgetsRender.go]: Maybe check all strings here as to not have [_white_space_][punctuation] such as ` : %d / %d` pattern in strings

File: cmd/node/main.go [hunk 3]
```diff
@@ -95,12 +95,6 @@ VERSION:
 		Usage: "The node will extract initial nodes info from the nodesSetup.json",
 		Value: "./config/nodesSetup.json",
 	}
-	// txSignSk defines a flag for the path of the single sign private key used when starting the node
```
Comment (iulianpascalau) [inline cmd/node/main.go]: Since we removed these flags, should also update the scripts from /scripts subpackage

File: node/options.go [hunk 4]
```diff
@@ -111,17 +111,6 @@ func WithDataStore(store dataRetriever.StorageService) Option {
 	}
 }
 
-// WithTxSignPrivKey sets up the single sign private key option for the Node
-func WithTxSignPrivKey(sk crypto.PrivateKey) Option {
```
Comment (SebastianMarian) [inline node/options.go]: You can also remove WithInitialNodesBalances from here, from tests and the n.initialNodesBalances variable from node.go

File: node/options_test.go [hunk 5]
```diff
@@ -172,18 +172,6 @@ func TestWithBlockChain_ShouldWork(t *testing.T) {
 	assert.Nil(t, err)
 }
 
-func TestWithDataStore_NilStoreShouldErr(t *testing.T) {
```
Comment (SebastianMarian) [inline node/options_test.go]: This should not be deleted. You need to fix the assert test and replace node.txSignPrivKey with the store variable

File: cmd/node/factory/structs.go [hunk 6]
```diff
@@ -2355,12 +2332,7 @@ func createMemUnit() storage.Storer {
 }
 
 // GetSigningParams returns a key generator, a private key, and a public key
-func GetSigningParams(
-	ctx *cli.Context,
-	skName string,
-	skIndexName string,
-	skPemFileName string,
-	suite crypto.Suite,
+func GetSigningParams(ctx *cli.Context, skName string, skIndexName string, skPemFileName string, suite crypto.Suite,
```
[the remark this answers is not in the document]
Comment (miiu96) [reply cmd/node/factory/structs.go]: done.

File: cmd/node/metrics/metrics.go [hunk 7]
```diff
@@ -78,8 +77,7 @@ func InitMetrics(
 }
 
 // SaveCurrentNodeNameAndPubKey will save metric in status handler with nodeName and transaction sign public key
-func SaveCurrentNodeNameAndPubKey(ash core.AppStatusHandler, txSignPk string, nodeName string) {
-	ash.SetStringValue(core.MetricPublicKeyTxSign, txSignPk)
+func SaveCurrentNodeNameAndPubKey(ash core.AppStatusHandler, nodeName string) {
```
[the remark this answers is not in the document]
Comment (miiu96) [reply cmd/node/metrics/metrics.go]: done.

File: node/nodeTesting.go (hunk 1 above)
Comment (miiu96) [reply node/nodeTesting.go]: done.

File: cmd/node/main.go [hunk 8]
```diff
@@ -95,12 +95,6 @@ VERSION:
...
-	txSignSk = cli.StringFlag{
```
[the remark this answers is not in the document]
Comment (miiu96) [reply cmd/node/main.go]: done.

File: cmd/node/main.go [hunk 9]
```diff
@@ -656,21 +642,16 @@ func startNode(ctx *cli.Context, log logger.Logger, version string) error {
...
 	if err != nil {
 		return err
 	}
 
-	txSignPk := factory.GetPkEncoded(cryptoComponents.TxSignPubKey)
-	metrics.SaveCurrentNodeNameAndPubKey(coreComponents.StatusHandler, txSignPk, preferencesConfig.Preferences.NodeDisplayName)
+	metrics.SaveCurrentNodeNameAndPubKey(coreComponents.StatusHandler, preferencesConfig.Preferences.NodeDisplayName)
 
-	sessionInfoFileOutput := fmt.Sprintf("%s:%s\n%s:%s\n%s:%s\n%s:%v\n%s:%s\n%s:%v\n",
+	sessionInfoFileOutput := fmt.Sprintf("%s:%s\n%s:%s\n%s\n:%v\n%s:%s\n%s:%v\n",
```
[the remark this answers is not in the document]
Comment (miiu96) [reply cmd/node/main.go]: done.

File: cmd/node/metrics/metrics.go [hunk 10]
```diff
@@ -78,8 +77,7 @@ func InitMetrics(
 }
 
 // SaveCurrentNodeNameAndPubKey will save metric in status handler with nodeName and transaction sign public key
```
[the remark this answers is not in the document]
Comment (miiu96) [reply cmd/node/metrics/metrics.go]: done.

File: node/options.go (hunk 4 above)
Comment (miiu96) [reply node/options.go]: done.

File: cmd/node/main.go (hunk 3 above)
Comment (miiu96) [reply cmd/node/main.go]: done.

File: node/options_test.go (hunk 5 above)
Comment (miiu96) [reply node/options_test.go]: done.

File: statusHandler/view/termuic/termuiRenders/widgetsRender.go (hunk 2 above)
Comment (miiu96) [reply statusHandler/view/termuic/termuiRenders/widgetsRender.go]: done.

File: scripts/testnet/include/nodes.sh [hunk 11]
```diff
@@ -123,12 +123,10 @@ assembleCommand_startObserverNode() {
   OBSERVER_INDEX=$2
   let "PORT = $PORT_ORIGIN_OBSERVER + $OBSERVER_INDEX"
   let "RESTAPIPORT=$PORT_ORIGIN_OBSERVER_REST + $OBSERVER_INDEX"
-  let "KEY_INDEX=$TOTAL_NODECOUNT - $OBSERVER_INDEX - 1"
   WORKING_DIR=$TESTNETDIR/node_working_dirs/observer$OBSERVER_INDEX
 
   local nodeCommand="./node \
         -port $PORT -log-save -log-level $LOGLEVEL -rest-api-interface localhost:$RESTAPIPORT \
-        -tx-sign-sk-index $KEY_INDEX -sk-index $KEY_INDEX \
```
Comment (iulianpascalau) [inline scripts/testnet/include/nodes.sh]: Should revert the `-sk-index $KEY_INDEX \` part as this is used for the block sign key

File: scripts/testnet/include/nodes.sh [hunk 12]
```diff
@@ -149,12 +147,10 @@ assembleCommand_startValidatorNode() {
   VALIDATOR_INDEX=$1
   let "PORT = $PORT_ORIGIN_VALIDATOR + $VALIDATOR_INDEX"
   let "RESTAPIPORT=$PORT_ORIGIN_VALIDATOR_REST + $VALIDATOR_INDEX"
-  let "KEY_INDEX=$VALIDATOR_INDEX"
   WORKING_DIR=$TESTNETDIR/node_working_dirs/validator$VALIDATOR_INDEX
 
   local nodeCommand="./node \
         -port $PORT -log-save -log-level $LOGLEVEL -rest-api-interface localhost:$RESTAPIPORT \
-        -tx-sign-sk-index $KEY_INDEX -sk-index $KEY_INDEX \
```
Comment (iulianpascalau) [inline scripts/testnet/include/nodes.sh]: Also here: Should revert the `-sk-index $KEY_INDEX \` part as this is used for the block sign key

File: scripts/testnet/include/nodes.sh (hunk 11 above)
Comment (miiu96) [reply scripts/testnet/include/nodes.sh]: done.

File: scripts/testnet/include/nodes.sh (hunk 12 above)
Comment (miiu96) [reply scripts/testnet/include/nodes.sh]: done.

**Pull request merged**